imap_processing/lo/l1b/lo_l1b.py

```diff
@@ -1254,8 +1254,10 @@ def resweep_histogram_data(
     o_counts_reswept = np.zeros_like(l1b_histrates["o_counts"].values)
 
     # Get the number of azimuth bins from the l1b_histrates dataset
-    num_azimuth = l1b_histrates["h_counts"].shape[1]
-    exposure_factor = np.zeros((len(epochs), num_azimuth, 7), dtype=int)
+    num_azimuth = l1b_histrates["h_counts"].shape[2]
```

Comment:
You could change this to access the dimension size directly:
```suggestion
    num_azimuth = l1b_histrates.sizes["spin_bin_6"]
```